src/uriHandler.test.ts

```diff
@@ -0,0 +1,85 @@
+import * as vscode from "vscode";
+import { UriEventHandler } from "../src/uriHandler";
+import { ScaffoldV1Template } from "../src/clients/scaffoldingService";
+import * as applyTemplateModule from "../src/scaffold";
+import * as sinon from "sinon";
+import assert from "assert";
+
+describe("UriEventHandler - /projectScaffold", () => {
+  let uriHandler: UriEventHandler;
+  let applyTemplateStub: sinon.SinonStub;
+
+  beforeEach(() => {
+    uriHandler = UriEventHandler.getInstance();
+    applyTemplateStub = sinon.stub(applyTemplateModule, "applyTemplate");
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
```

Comment:
Just to keep stubs isolated so we don't inadvertently affect other test suites (even though there's the `sinon.restore()`:
```suggestion
describe("UriEventHandler - /projectScaffold", () => {
  let sandbox: sinon.SinonSandbox;
  let uriHandler: UriEventHandler;
  let applyTemplateStub: sinon.SinonStub;

  beforeEach(() => {
    sandbox = sinon.createSandbox();
    applyTemplateStub = sandbox.stub(applyTemplateModule, "applyTemplate");
    
    uriHandler = UriEventHandler.getInstance();
  });

  afterEach(() => {
    sandbox.restore();
  });
```